helion/_compiler/device_ir.py

```diff
@@ -259,8 +260,7 @@ def add_reduction_loop_graph(
         )
 
     def add_root_graph(self, graph: torch.fx.GraphModule) -> None:
-        assert self.root_id is None
```

Comment:
We should add an assert check that the device loops are consecutive.  E.g. the following should be an error:
```py
for a in hl.tile(A):
   ...
foo.zero_()
for b in hl.tile(B):
  ....
```
since in reality `foo.zero_()` cannot run between the two loops.